pageserver/src/tenant/storage_layer/inmemory_layer.rs

```diff
@@ -78,10 +78,10 @@ impl std::fmt::Debug for InMemoryLayer {
 }
 
 pub struct InMemoryLayerInner {
-    /// All versions of all pages in the layer are kept here.  Indexed
+    /// All versions of all pages in the layer are kept here. Indexed
     /// by block number and LSN. The value is an offset into the
     /// ephemeral file where the page version is stored.
-    index: HashMap<Key, VecMap<Lsn, u64>>,
+    index: BTreeMap<Key, VecMap<Lsn, u64>>,
```

Comment:
This makes a lot of sense for the read path, but we hand off the cost to the write path. I think we should try to quantify what that cost is (perhaps we're fine with it). The `test_bulk_insert` unit test and `test_bulk_insert` performance test would be a good place to start.

Reply:
Actually, putting a value to an in-memory layer will incur a disk write, and therefore I don't think the in-memory structure would be an actual bottleneck. I can run some experiments locally to see if the perf impacts.

Reply:
There's a buffer in between so it's not necessarily always true that we go to disk for every write. Would be good to check imo.